include/cvc5/c/cvc5.h

```diff
@@ -0,0 +1,3936 @@
+ * Determine if a given synthesis result is empty (a nullary result) and not an
+ * actual result returned from a synthesis query.
+ * @param result The result.
+ * @return True if the given result is a nullary result.
+ */
+bool cvc5_synth_result_is_null(const Cvc5SynthResult* result);
+
+/**
+ * Determine if the given result is from a synthesis query that has a solution.
+ * @param result The result.
+ * @return True if the synthesis query has a solution.
+ */
+bool cvc5_synth_result_has_solution(const Cvc5SynthResult* result);
+
+/**
+ * Determine if the given result is from a synthesis query that has no solution.
+ * @param result The result.
+ * @return True if the synthesis query has no solution. In this case, it
+ *         was determined that there was no solution.
+ */
+bool cvc5_synth_result_has_no_solution(const Cvc5SynthResult* result);
+
+/**
+ * Determine if the given result is from a synthesis query where its result
+ * could not be determined.
+ * @param result The result.
+ * @return True if the result of the synthesis query could not be determined.
+ */
+bool cvc5_synth_result_is_unknown(const Cvc5SynthResult* result);
+
+/**
+ * Get the string representation of a given result.
+ * @param result The result.
+ * @return A string representation of the given synthesis result.
+ */
+const char* cvc5_synth_result_to_string(const Cvc5SynthResult* result);
+
+
+/* -------------------------------------------------------------------------- */
+/* Cvc5Sort                                                                   */
+/* -------------------------------------------------------------------------- */
+
+/**
+ * Compare two sorts for structural equality.
+ * @param a The first sort.
+ * @param b The second sort.
+ * @return True if the sorts are equal.
+ */
+bool cvc5_sort_is_equal(Cvc5Sort a, Cvc5Sort b);
+
+/**
+ * Compare two sorts for structural disequality.
+ * @param a The first sort.
+ * @param b The second sort.
+ * @return True if the sorts are not equal.
+ */
+bool cvc5_sort_is_disequal(Cvc5Sort a, Cvc5Sort b);
+
+/**
+ * Get the kind of the given sort.
+ * @param sort The sort.
+ * @return The kind of the sort.
+ *
+ * @warning This function is experimental and may change in future versions.
+ */
+Cvc5SortKind cvc5_sort_get_kind(Cvc5Sort sort);
+
+/**
+   * Determine if the given sort has a symbol (a name).
+ *
+ * For example, uninterpreted sorts and uninterpreted sort constructors have
+ * symbols.
+ *
+ * @param sort The sort.
+ * @return True if the sort has a symbol.
+ */
+bool cvc5_sort_has_symbol(Cvc5Sort sort);
+
+/**
+ * Get the symbol of this Sort.
+ *
+ * @note Asserts cvc5_sort_has_symbol().
+ *
+ * The symbol of this sort is the string that was
+ * provided when constructing it via
+ * cvc5_mk_uninterpreted_sort(),
+ * cvc5_mk_unresolved_sort(), or
+ * cvc5_mk_uninterpreted_sort_constructor_sort().
+ *
+ * @param sort The sort.
+ * @return The raw symbol of the sort.
+ */
+const char* cvc5_sort_get_symbol();
```

Comment:
I'm guessing that if you return const char*, the user is not responsible for freeing the memory.  In such cases (here and below), who will free it?

Reply:
It will be static thread_local, entirely handled on the C++ level (the implementation of the C API is in C++). See for example https://github.com/bitwuzla/bitwuzla/blob/main/src/api/c/bitwuzla.cpp#L116

Reply:
Ok sounds good.